designcompose/src/main/java/com/android/designcompose/Layout.kt

```diff
@@ -575,9 +591,23 @@ internal fun designMeasurePolicy(view: View, layoutId: Int) =
             Log.d(TAG, "designMeasurePolicy error: null layout $name layoutId $layoutId")
         // Get width and height from constraints if they are fixed. Otherwise get them from layout.
         val myWidth =
-            if (constraints.hasFixedWidth) constraints.maxWidth else myLayout?.width() ?: 0
+            if (constraints.hasFixedWidth && constraints.maxWidth != 0) constraints.maxWidth
+            else myLayout?.width() ?: 0
         val myHeight =
-            if (constraints.hasFixedHeight) constraints.maxHeight else myLayout?.height() ?: 0
+            if (constraints.hasFixedHeight && constraints.maxHeight != 0) constraints.maxHeight
+            else myLayout?.height() ?: 0
+
+        // If constraints have forced a size that does not match our layout size, set this size into
```

Comment:
Oh neat - so `Modifier.fillMaxSize` etc just work directly because they set `hasFixed{Width|Height}` and `max{Width|Height}`? That's cool!